src/components/countdown/Countdown.tsx

```diff
@@ -0,0 +1,65 @@
+import { useState, useEffect } from "react"
+
+interface CountdownProps {
+  targetDate: Date
+}
+
+const Countdown: React.FC<CountdownProps> = ({ targetDate }) => {
+  const calculateTimeLeft = () => {
+    const difference = +targetDate - +new Date()
+    let timeLeft = { days: 0, hours: 0, minutes: 0 }
+
+    if (difference > 0) {
+      timeLeft = {
+        days: Math.floor(difference / (1000 * 60 * 60 * 24)),
+        hours: Math.floor((difference / (1000 * 60 * 60)) % 24),
+        minutes: Math.floor((difference / 1000 / 60) % 60),
+      }
+    }
+
+    return timeLeft
+  }
+
+  const [timeLeft, setTimeLeft] = useState(calculateTimeLeft())
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setTimeLeft(calculateTimeLeft())
+    }, 1000)
+
+    return () => clearTimeout(timer)
+  }, [timeLeft])
+
+  return (
+    <div className="flex items-center justify-center space-x-1">
+      <div className="flex">
```

Comment:
these outer divs seem to be unnecessary, maybe we can get rid of them?

Reply:
Im afraid it is necessary